athena-vertica/src/main/java/com/amazonaws/athena/connectors/vertica/VerticaMetadataHandler.java

```diff
@@ -243,20 +316,29 @@ public void getPartitions(BlockWriter blockWriter, GetTableLayoutRequest request
 
         //Build the SQL query
         Connection connection = getConnection(request);
-        DatabaseMetaData dbMetadata = connection.getMetaData();
-        ResultSet definition = dbMetadata.getColumns(null, tableName.getSchemaName(), tableName.getTableName(), null);
+
+        // if  QPT get input query from Athena console
+        //else old logic
 
         VerticaExportQueryBuilder queryBuilder = queryFactory.createVerticaExportQueryBuilder();
+        String preparedSQLStmt;
 
-        String preparedSQLStmt = queryBuilder.withS3ExportBucket(s3ExportBucket)
-                .withQueryID(queryID)
-                .withColumns(definition, schemaName)
-                .fromTable(tableName.getSchemaName(), tableName.getTableName())
-                .withConstraints(constraints, schemaName)
-                .build();
+        if (!request.getTableName().getQualifiedTableName().equalsIgnoreCase(queryPassthrough.getFunctionSignature())) {
```

Comment:
I think a better way to check if a request is a query passthrough request is using: `request.getConstraints().isQueryPassThrough()`